### PR TITLE
[Enhencement](Jdbc catalog) Add two optional properties for jdbc catalog

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #xxx

## Problem summary
1. The first property is `only_specified_database`:
In the past, `Jdbc Catalog` will synchronize all database from source database.
Now we add a parameter called `only_specified_database` to jdbc catalog to allow only the specified database to be synchronized, eg:

```sql
create resource if not exists ${resource_name} properties(
    "type"="jdbc",
    "user"="root",
    "password"="123456",
    "jdbc_url" = "jdbc:mysql://172.18.0.1:${mysql_port}/doris_test?useSSL=false",
    "driver_url" = "https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/jdbc_driver/mysql-connector-java-8.0.25.jar",
    "driver_class" = "com.mysql.cj.jdbc.Driver",
    "only_specified_database" = "true"
);
```
if `only_specified_database` is `true`, jdbc catalog will only synchronize the database which is specified in `jdbc_url`.

2. The second property is `lower_case_table_names`:
This property will synchronize jdbc external data source table names in lower case.

```sql
create resource if not exists ${resource_name} properties(
  "type"="jdbc",
  "user"="doris_test",
  "password"="123456",
  "jdbc_url" = "jdbc:oracle:thin:@172.18.0.1:${oracle_port}:${SID}",
  "driver_url" = "https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/jdbc_driver/ojdbc8.jar",
  "driver_class" = "oracle.jdbc.driver.OracleDriver",
  "lower_case_table_names" = "true"
);
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/docs/lakehouse/multi-catalog/jdbc.md [hunk 1]
```diff
@@ -176,6 +164,7 @@ Currently, Jdbc Catalog only support to use 5.x version of JDBC jar package to c
 | `jdbc_url `     | Yes             |               | JDBC connection string                             |
 | `driver_url `   | Yes             |               | JDBC Driver Jar                                    |
 | `driver_class ` | Yes             |               | JDBC Driver Class                                  |
+| <version since="dev">`only_specified_database`</version> | No             |     "false"          | When the JDBC is connected, you can specify which database/schema to connect. For example, you can specify the DataBase in mysql `jdbc_url`; you can specify the CurrentSchema in PG `jdbc_url`, `only_specified_database` specifies whether only the database is specified simultaneously.                                  |
```
Comment (morningman) [inline docs/en/docs/lakehouse/multi-catalog/jdbc.md:167]: do not support this type of `<version>`
you have to explain it below, as `driver_url`

File: docs/zh-CN/docs/lakehouse/multi-catalog/jdbc.md [hunk 2]
```diff
@@ -173,9 +161,10 @@ CREATE CATALOG doris_catalog PROPERTIES (
 --- | --- | --- | --- 
 `user` | 是 | | 对应数据库的用户名 |
 `password` | 是 |   | 对应数据库的密码 |
-`jdbc_url ` | 是 |  | JDBC 连接串 |
-`driver_url ` | 是 |  | JDBC Driver Jar 包名称* |
-`driver_class ` | 是 |  | JDBC Driver Class 名称 |
+`jdbc_url` | 是 |  | JDBC 连接串 |
+`driver_url` | 是 |  | JDBC Driver Jar 包名称* |
+`driver_class` | 是 |  | JDBC Driver Class 名称 |
+<version since="dev">`only_specified_database`</version> | 否 | "false" | 在jdbc连接时可以指定链接到哪个database/schema, 如：mysql中jdbc_url中可以指定database, pg的jdbc_url中可以指定currentSchema，`only_specified_database`指定是否只同步指定的 database.  |
```
Comment (morningman) [inline docs/zh-CN/docs/lakehouse/multi-catalog/jdbc.md:167]: ditto

File: docs/en/docs/lakehouse/multi-catalog/jdbc.md (hunk 1 above)
Comment (morningman) [reply docs/en/docs/lakehouse/multi-catalog/jdbc.md:167]: And you forgot the add `lower_case_table_names` doc

File: fe/fe-core/src/main/java/org/apache/doris/catalog/JdbcResource.java [hunk 3]
```diff
@@ -144,22 +156,46 @@ public void checkProperties(Map<String, String> properties) throws AnalysisExcep
...
         this.configs.put(JDBC_URL, handleJdbcUrl(getProperty(JDBC_URL)));
         configs.put(CHECK_SUM, computeObjectChecksum(getProperty(DRIVER_URL)));
     }
 
+    /**
+     * This function used to handle optional arguments
+     * eg: only_specified_database、lower_case_table_names
+     */
+    private void handleOptionalArguments() {
+        for (String s : OPTIONAL_PROPERTIES) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/JdbcResource.java:192]: Not all optional properties using `false` as default.
So better use `if else` or `switch` to handle them separately

File: docs/zh-CN/docs/lakehouse/multi-catalog/jdbc.md (hunk 2 above)
Comment (BePPPower) [reply docs/zh-CN/docs/lakehouse/multi-catalog/jdbc.md]: done

File: fe/fe-core/src/main/java/org/apache/doris/catalog/JdbcResource.java (hunk 3 above)
Comment (BePPPower) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/JdbcResource.java:200]: done

File: docs/en/docs/lakehouse/multi-catalog/jdbc.md (hunk 1 above)
Comment (BePPPower) [reply docs/en/docs/lakehouse/multi-catalog/jdbc.md]: done

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.52 seconds
 stream load tsv:          481 seconds loaded 74807831229 Bytes, about 148 MB/s
 stream load json:         37 seconds loaded 2358488459 Bytes, about 60 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          27 seconds loaded 861443392 Bytes, about 30 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230301153024_clickbench_pr_106608.html

**Comment by BePPPower** [comment]
run feut

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**